CMP-3807: Added TestCSVInfrastructureFeaturesAnnotation to cover annotation for disconnected environments.

taimurhafeez opened the pull request:
It tests that the Compliance Operator CSV has the proper infrastructure features annotation for disconnected environments.
Since it requires OLM based installation, I used [PR-1001](https://github.com/ComplianceAsCode/compliance-operator/pull/1001). The combined PR is in draft and available here [PR-1039](https://github.com/ComplianceAsCode/compliance-operator/pull/1039). 
expected output:
on OCP 4.22: 
```
E2E_INSTALL_METHOD=subscription make e2e-parallel E2E_GO_TEST_FLAGS="-v -run TestCSVInfrastructureFeaturesAnnotation"
cd config/manager && /home/thafeez/repos/CaC/compliance-operator/build/_tools/kustomize edit set image compliance-operator=ghcr.io/complianceascode/compliance-operator:latest
rm -rf tests/_setup
mkdir -p tests/_setup
/home/thafeez/repos/CaC/compliance-operator/build/_tools/kustomize build config/no-ns | sed -e 's%ghcr.io/complianceascode/compliance-operator:latest%ghcr.io/complianceascode/compliance-operator:latest%' -e 's%ghcr.io/complianceascode/k8scontent:latest%ghcr.io/complianceascode/k8scontent:latest%' -e 's%ghcr.io/complianceascode/openscap-ocp:latest%ghcr.io/complianceascode/openscap-ocp:latest%'  > tests/_setup/deploy_rbac.yaml
/home/thafeez/repos/CaC/compliance-operator/build/_tools/kustomize build config/crd > tests/_setup/crd.yaml
2026/05/07 11:12:02 switching to /home/thafeez/repos/CaC/compliance-operator directory to setup and execute tests
2026/05/07 11:12:03 creating namespace osdk-e2e-288a5a67-bfda-481d-a065-454d59cb254d
2026/05/07 11:12:03 Registering OLM schemes for subscription-based installation
2026/05/07 11:12:05 Installing compliance operator via OLM Subscription
2026/05/07 11:12:05 creating OperatorGroup osdk-e2e-288a5a67-bfda-481d-a065-454d59cb254d in namespace osdk-e2e-288a5a67-bfda-481d-a065-454d59cb254d
2026/05/07 11:12:05 creating Subscription compliance-operator in namespace osdk-e2e-288a5a67-bfda-481d-a065-454d59cb254d
2026/05/07 11:12:05 Waiting for operator installation via OLM...
2026/05/07 11:12:11 Compliance Operator CSV not found yet, waiting...
2026/05/07 11:12:16 Compliance Operator CSV not found yet, waiting...
2026/05/07 11:12:21 Compliance Operator CSV not found yet, waiting...
2026/05/07 11:12:26 Compliance Operator CSV not found yet, waiting...
2026/05/07 11:12:31 CSV compliance-operator.v1.9.0 is in phase Pending, waiting...
2026/05/07 11:12:36 CSV compliance-operator.v1.9.0 is in phase Pending, waiting...
2026/05/07 11:12:41 CSV compliance-operator.v1.9.0 is in phase Installing, waiting...
2026/05/07 11:12:46 CSV compliance-operator.v1.9.0 is in Succeeded phase
2026/05/07 11:12:46 Operator CSV installed successfully
2026/05/07 11:12:46 Waiting for compliance operator CRDs to be installed by OLM
2026/05/07 11:12:46 Waiting for compliance operator CRDs to be created by OLM (timeout: 5m0s)
2026/05/07 11:12:51 CRD compliancescans.compliance.openshift.io is available
2026/05/07 11:12:57 CRD compliancesuites.compliance.openshift.io is available
2026/05/07 11:13:02 CRD complianceremediations.compliance.openshift.io is available
2026/05/07 11:13:07 CRD scansettings.compliance.openshift.io is available
2026/05/07 11:13:13 CRD scansettingbindings.compliance.openshift.io is available
2026/05/07 11:13:18 CRD tailoredprofiles.compliance.openshift.io is available
2026/05/07 11:13:23 CRD profiles.compliance.openshift.io is available
2026/05/07 11:13:28 CRD profilebundles.compliance.openshift.io is available
2026/05/07 11:13:33 CRD rules.compliance.openshift.io is available
2026/05/07 11:13:39 CRD variables.compliance.openshift.io is available
2026/05/07 11:13:44 CRD compliancecheckresults.compliance.openshift.io is available
2026/05/07 11:13:44 All compliance operator CRDs are available
2026/05/07 11:13:44 registering resource schemes with framework
2026/05/07 11:14:06 Deployment available (1/1)
2026/05/07 11:14:11 ProfileBundle ready (VALID)
2026/05/07 11:14:16 ProfileBundle ready (VALID)
2026/05/07 11:14:19 adding label node-role.kubernetes.io/e2e to node ip-10-0-115-173.us-west-1.compute.internal
2026/05/07 11:14:19 e2e Machine Config Pool has not updated... retrying
2026/05/07 11:14:30 e2e Machine Config Pool has not updated... retrying
2026/05/07 11:14:40 e2e Machine Config Pool has not updated... retrying
2026/05/07 11:14:50 successfully created Machine Config Pool e2e
2026/05/07 11:14:50 ValidatingAdmissionPolicy 'custom-machine-config-pool-selector' exists, adding minimal selectors to MachineConfigPool e2e-invalid
=== RUN   TestCSVInfrastructureFeaturesAnnotation
=== PAUSE TestCSVInfrastructureFeaturesAnnotation
=== CONT  TestCSVInfrastructureFeaturesAnnotation
    main_test.go:5698: OLM is available on the cluster
    main_test.go:5731: Found compliance-operator CSV: compliance-operator.v1.9.0
    main_test.go:5754: CSV compliance-operator.v1.9.0 correctly has infrastructure-features annotation: ["disconnected", "fips", "proxy-aware"]
--- PASS: TestCSVInfrastructureFeaturesAnnotation (0.93s)
PASS
```

xiaojiey committed:
Add subscription-based operator installation

Address review comments for subscription-based installation

- Add waitForOperatorInstallation() to poll for CSV readiness after OLM subscription
  This ensures the operator and CRDs are ready before proceeding with test setup
- Make manifest cleanup conditional based on installation method
  Only clean up YAML manifests if installed via manifests, not subscription

taimurhafeez committed:
Added TestCSVInfrastructureFeaturesAnnotation to cover downstream test 40280. It tests that the Compliance Operator CSV
has the proper infrastructure features annotation for disconnected environments.

taimurhafeez committed:
Removed test ID

taimurhafeez committed:
Removed test ID

rhmdnd reviewed (changes requested):
We might need to provision a new CI job in openshift/release that leverages @xiaojiey's patch to install from subscriptions to run this. Otherwise, this will skip until we have that in place I think.

One recommendation on the assertion structure. Otherwise the test looks great (just need to think about how we're going to wire this up).

taimurhafeez committed:
Parse infrastructure-features annotation as JSON

Replace exact string match with JSON parsing for more robust
validation and better error messages.

taimurhafeez committed:
Parse infrastructure-features annotation as JSON

Replace exact string match with JSON parsing for more robust
validation and better error messages.

rhmdnd commented on tests/e2e/parallel/main_test.go:
```diff
@@ -5434,86 +5311,79 @@ func TestRuleVariableAnnotation(t *testing.T) {
+	for i := range csvList.Items {
+		name := csvList.Items[i].GetName()
+		// CSV names typically start with "compliance-operator"
+		if len(name) >= 19 && name[:19] == "compliance-operator" {
```
We're not doing an exactly string check here because version information is baked in to the CSV name, which would this test fragile with each release. Is that correct?

taimurhafeez replied:
yes exactly, that was the intention here to match just **_compliance-operator_**

rhmdnd reviewed (approved):
/lgtm

rhmdnd commented:
This will need a rebase and a clean CI run, but the overall test change looks straightforward and makes sense.

taimurhafeez committed:
removed uncessary whitespaces incurred in previous commit

taimurhafeez committed:
Fix rebase artifacts in parallel e2e tests

Restore TestCELWithXCCDFProfileScan, TestMultipleProfileBundlesWithTailoredProfiles,
and TestCSVInfrastructureFeaturesAnnotation that were mangled during rebase.
Rename misnamed TestTimeoutDisabledWithZeroValue back to its correct name.
